### PR TITLE
Add and fix docs on consensus modules

### LINKED ISSUE #54: Document consensus-related modules
The network simulator, simple node and Nakamoto node need haddocks, as well as the consensus-related type classes.

- [ ] Document BlockHeader and the `s` parameter.

### PR DESCRIPTION
**Pull request by cloudhead**
Closes #54

### CODE REVIEW

File: src/Oscoin/Consensus.hs [hunk 1]
```diff
@@ -34,8 +34,8 @@ validateBlockchain validateBlock (Blockchain (blk :| blks)) =
     validateBlock blks blk *> validateBlockchain validateBlock (fromList blks)
 
 
--- | Validates the size (in bytes) for a block, comparing the serialised size
--- with the maximum allowed value as read from the configuration file.
+-- | Validates the size (in bytes) of a block, comparing the serialized size
```
Comment (adinapoli-mndc) [inline src/Oscoin/Consensus.hs]: so are we standardising on US spelling? :trollface:

File: src/Oscoin/Consensus/Nakamoto.hs [hunk 2]
```diff
@@ -171,14 +179,14 @@ difficulty
 difficulty = encodeDifficulty . os2ip . headerHash
 
 -- | Number of blocks to consider for difficulty calculation. Roughly 2 weeks
--- withh a 10 min block time.
+-- with a 10 minute block time.
```
Comment (adinapoli-mndc) [inline src/Oscoin/Consensus/Nakamoto.hs]: 10 _minutes_ perhaps?

File: src/Oscoin/Consensus.hs (hunk 1 above)
Comment (cloudhead) [reply src/Oscoin/Consensus.hs]: Yeah aka "International English" :thinking:

File: src/Oscoin/Consensus/Nakamoto.hs (hunk 2 above)
Comment (cloudhead) [reply src/Oscoin/Consensus/Nakamoto.hs]: But it's '**a** 10 minute block time'

Comment (adinapoli-mndc) [reply src/Oscoin/Consensus/Nakamoto.hs]: oh, right, right, sorry 'bout that 😁

**Pull request merged**